languages/nodejs/repl.js

```diff
@@ -145,20 +146,23 @@ if (process.env.PRYBAR_CODE) {
     root: global,
   };
 
-  // These properties will show a warning. We can avoid them.
-  for (const prop of Object.getOwnPropertyNames(global)) {
-    if (sandbox.hasOwnProperty(prop)) {
-      continue;
-    }
-
-    sandbox[prop] = global[prop];
-  }
-
   console.log(
     "\u001b[0m\u001b[90mHint: hit control+c anytime to enter REPL.\u001b[0m"
   );
   const context = vm.createContext(sandbox);
 
+  // Adapted from https://bit.ly/2Fc4WjM
+  const globalBuiltins =
+  new Set(vm.runInNewContext('Object.getOwnPropertyNames(globalThis)'));
+  for (const name of Object.getOwnPropertyNames(global)) {
+    // Only set properties that do not already exist as a global builtin.
+    if (!globalBuiltins.has(name)) {
+      Object.defineProperty(context, name,
+                           Object.getOwnPropertyDescriptor(global, name));
+    }
+  }
+  context.global = context;
```

Comment:
Ideally we make this non-writable.

Reply:
It makes sense to set this before the loop.
Also on line 144 and 145 we set the `context.root` and `context.GLOBAL` to equal this script's global, we should remove them.

Node 12 also adds `globalThis` which [seems to be the future of all global objects ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis)

Here's something i wrote before, it includes deprecation notices.

```suggestion
const util = require('util');

// VM context doesn't have a built-in "global", so we
// have to tell it to point to itself
Object.defineProperties(context, {
  global: {
    ...Object.getOwnPropertyDescriptor(global, 'global'),
    value: context,
  },
  globalThis: {
    ...Object.getOwnPropertyDescriptor(global, 'globalThis'),
    value: context,
  },
  GLOBAL: {
    ...Object.getOwnPropertyDescriptor(global, 'GLOBAL'),
    get: util.deprecate(() => context, `'GLOBAL' is deprecated, use 'global'`, 'DEP0016'),
    set: util.deprecate(
      (value) => {
        Object.defineProperty(context, 'GLOBAL', {
          configurable: true,
          writable: true,
          enumerable: true,
          value: value,
        });
      },
      `'GLOBAL' is deprecated, use 'global'`,
      'DEP0016',
    ),
  },
  root: {
    ...Object.getOwnPropertyDescriptor(global, 'root'),
    get: util.deprecate(() => context, `'root' is deprecated, use 'global'`, 'DEP0016'),
    set: util.deprecate(
      (value) => {
        Object.defineProperty(context, 'root', {
          configurable: true,
          writable: true,
          enumerable: true,
          value: value,
        });
      },
      `'root' is deprecated, use 'global'`,
      'DEP0016',
    ),
  },
});
```

Reply:
I also think it would be better to do this before the loop for defining other properties